CMakeLists.txt

```diff
@@ -422,46 +422,60 @@ endif()
 
 # Setup Ginkgo
 if (PRECICE_FEATURE_GINKGO_MAPPING)
+  add_library(ginkgoKernels STATIC)
+  set_target_properties(ginkgoKernels PROPERTIES
+    CXX_STANDARD 17
+    CXX_STANDARD_REQUIRED Yes
+    CXX_EXTENSIONS No
+    POSITION_INDEPENDENT_CODE YES
+    )
     # Next we configure the common kernels for the matrix assembly
-    target_sources(preciceCore PRIVATE
+    target_sources(ginkgoKernels PRIVATE
     src/mapping/device/GinkgoRBFKernels.cpp
     src/mapping/device/Ginkgo.cpp
     )
 
-  target_link_libraries(preciceCore PRIVATE Ginkgo::ginkgo Kokkos::kokkos)
-  target_compile_definitions(preciceCore PUBLIC EIGEN_NO_CUDA)
+    target_include_directories(ginkgoKernels PUBLIC
+      $<BUILD_INTERFACE:${preCICE_SOURCE_DIR}/src>
+      $<BUILD_INTERFACE:${preCICE_BINARY_DIR}/src>
+    )
+
+  target_link_libraries(ginkgoKernels PRIVATE Ginkgo::ginkgo Kokkos::kokkos fmtlib-static)
+  target_compile_definitions(ginkgoKernels PUBLIC EIGEN_NO_CUDA)
+
+  target_link_libraries(preciceCore PUBLIC ginkgoKernels)
 
   # Disable compiler launcher since Kokkos ships its own nvcc wrapper
-   set_target_properties(preciceCore PROPERTIES CXX_COMPILER_LAUNCHER "")
+   set_target_properties(ginkgoKernels PROPERTIES CXX_COMPILER_LAUNCHER "")
 
   # Now we setup cuSolver and hipSOLVER for preCICE
   if(PRECICE_WITH_CUDA)
-    target_compile_definitions(preciceCore PUBLIC PRECICE_WITH_CUDA)
+    target_compile_definitions(ginkgoKernels PUBLIC PRECICE_WITH_CUDA)
     # Required to make nvcc happy
-    target_compile_definitions(preciceCore PRIVATE BOOST_PP_VARIADICS=1)
-    target_sources(preciceCore PRIVATE ${PROJECT_SOURCE_DIR}/src/mapping/device/CudaQRSolver.cuh ${PROJECT_SOURCE_DIR}/src/mapping/device/CudaQRSolver.cu)
-    target_link_libraries(preciceCore PRIVATE cusolver)
+    target_compile_definitions(ginkgoKernels PRIVATE BOOST_PP_VARIADICS=1)
+    target_sources(ginkgoKernels PRIVATE ${PROJECT_SOURCE_DIR}/src/mapping/device/CudaQRSolver.cuh ${PROJECT_SOURCE_DIR}/src/mapping/device/CudaQRSolver.cu)
+    target_link_libraries(ginkgoKernels PRIVATE cusolver)
```

Comment:
Can you rebase this on develop? I think `cublas` and `hipblas` are missing here. https://github.com/precice/precice/pull/2079 should give a conflict

Reply:
Fixed